packages/@livestore/common/src/leader-thread/LeaderSyncProcessor.ts

```diff
@@ -303,15 +311,15 @@ export const makeLeaderSyncProcessor = ({
       ) =>
         Effect.gen(function* () {
           if (onError === 'ignore') {
-            if (LS_DEV) {
-              yield* Effect.logDebug(
-                `Ignoring sync error (${cause._tag === 'Fail' ? cause.error._tag : cause._tag})`,
-                Cause.pretty(cause),
-              )
-            }
+            yield* Effect.logDebug(
+              `Ignoring sync error (${cause._tag === 'Fail' ? cause.error._tag : cause._tag})`,
+              Cause.pretty(cause),
+            )
```

Comment:
IMO ignoring the error here may not be the best approach, I was having trouble while declaring the sync backend, I was getting lots of errors on my end and I didn't see much - e.g. I defined the response schema wrong and it was throwing parse error wrapped into unknown error and no logs. Same for the below thread crash. 

Besides logging, not sure why would we ignore this because lets say my reponse schema is wrong, which is unrecoverable, so e.g. backend pulling process will permanently crash and only page refresh will try it again. I would just let it crash completely and log error.